src/v/cloud_storage/topic_manifest.cc

```diff
@@ -42,68 +206,24 @@ ss::future<> topic_manifest::update(ss::input_stream<char> is) {
-    json::Document m;
-    rapidjson::IStreamWrapper wrapper(stream);
-    m.ParseStream(wrapper);
-    update(m);
-    co_return;
-}
 
-void topic_manifest::update(const json::Document& m) {
-    auto ver = m["version"].GetInt();
-    if (ver != static_cast<int>(topic_manifest_version::v1)) {
-        throw std::runtime_error("topic manifest version not supported");
-    }
-    auto ns = model::ns(m["namespace"].GetString());
-    auto tp = model::topic(m["topic"].GetString());
-    int32_t partitions = m["partition_count"].GetInt();
-    int16_t rf = m["replication_factor"].GetInt();
-    int32_t rev = m["revision_id"].GetInt();
-    cluster::topic_configuration conf(ns, tp, partitions, rf);
-    // optional
-    if (!m["compression"].IsNull()) {
-        conf.properties.compression = boost::lexical_cast<model::compression>(
-          m["compression"].GetString());
-    }
-    if (!m["cleanup_policy_bitflags"].IsNull()) {
-        conf.properties.cleanup_policy_bitflags
-          = boost::lexical_cast<model::cleanup_policy_bitflags>(
-            m["cleanup_policy_bitflags"].GetString());
-    }
-    if (!m["compaction_strategy"].IsNull()) {
-        conf.properties.compaction_strategy
-          = boost::lexical_cast<model::compaction_strategy>(
-            m["compaction_strategy"].GetString());
-    }
-    if (!m["timestamp_type"].IsNull()) {
-        conf.properties.timestamp_type
-          = boost::lexical_cast<model::timestamp_type>(
-            m["timestamp_type"].GetString());
-    }
-    if (!m["segment_size"].IsNull()) {
-        conf.properties.segment_size = m["segment_size"].GetInt64();
-    }
-    // tristate
-    if (m.HasMember("retention_bytes")) {
-        if (!m["retention_bytes"].IsNull()) {
-            conf.properties.retention_bytes = tristate<size_t>(
-              m["retention_bytes"].GetInt64());
-        } else {
-            conf.properties.retention_bytes = tristate<size_t>(std::nullopt);
-        }
-    }
-    if (m.HasMember("retention_duration")) {
-        if (!m["retention_duration"].IsNull()) {
-            conf.properties.retention_duration
-              = tristate<std::chrono::milliseconds>(
-                std::chrono::milliseconds(m["retention_duration"].GetInt64()));
-        } else {
-            conf.properties.retention_duration
-              = tristate<std::chrono::milliseconds>(std::nullopt);
-        }
+    rapidjson::IStreamWrapper wrapper(stream);
+    rapidjson::Reader reader;
+    topic_manifest_handler handler;
+    if (reader.Parse(wrapper, handler)) {
+        vlog(cst_log.debug, "Parsed successfully!");
+        _rev = model::initial_revision_id(handler._revision_id);
+        _topic_config = handler._topic_config;
+    } else {
+        rapidjson::ParseErrorCode e = reader.GetParseErrorCode();
+        size_t o = reader.GetErrorOffset();
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format,
+          "Failed to parse topic manifest: {} at offset {}",
```

Comment:
Rule of thumb: all log messages/errors should include topic name/partition name for easy grepping. Topic name is included in the manifest path so we can use that.

Reply:
It's probably useful to know the manifest path as well.

Reply:
Added manifest path to this message